src/devices/Common/System/Device/Analog/TriggerReason.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Analog
+{
+    /// <summary>
+    /// Gives the reason why a new value was provided
+    /// </summary>
+    public enum TriggerReason
+    {
+        /// <summary>
+        /// The reason for the new message is unknown
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// A new value is available
+        /// </summary>
+        NewMeasurement,
+
+        /// <summary>
+        /// A new value is read with a specific frequency
+        /// </summary>
+        Timed,
+
+        /// <summary>
+        /// A value is provided when certain thresholds are exceeded
+        /// </summary>
+        LimitExceeded
```

Comment:
consider calling this `overflow`

Reply:
The "overflow" can be either direction, and overflow would indicate the value is to high (when it can also become to small)